src/SonarScanner.MSBuild.PreProcessor/JreResolution/JreResolver.cs

```diff
@@ -112,7 +112,11 @@ private bool IsValid(ProcessedArgs args)
             logger.LogDebug(Resources.MSG_JreResolver_ArchitectureMissing);
             return false;
         }
-
+        if (!server.SupportsJreProvisioning)
+        {
+            logger.LogDebug(Resources.MSG_JreResolver_NotSupportedByServer);
+            return false;
```

Comment:
Don't you think this could be a bit earlier?
I would probably put it between the `skip jre provisioning` and the `os missing` ones, as the first two are the user knowing what they are doing, but the other two are less "important" than needing to upgrade to SQ 10.6 to have this feature.

So priority wise, I would suggest:
- Java Exe path
- Skip Provisioning
- Server version check
- OS
- Architecture

WDYT?